pkg/ring/lifecycler.go

```diff
@@ -153,30 +153,38 @@ func NewLifecycler(cfg LifecyclerConfig, flushTransferer FlushTransferer) (*Life
 	return l, nil
 }
 
-// IsReady is used to rate limit the number of ingesters that can be coming or
+// CheckReady is used to rate limit the number of ingesters that can be coming or
 // going at any one time, by only returning true if all ingesters are active.
-func (i *Lifecycler) IsReady(ctx context.Context) bool {
+// The state latches: once we have gone ready we don't go un-ready
+func (i *Lifecycler) CheckReady(ctx context.Context) error {
 	i.readyLock.Lock()
 	defer i.readyLock.Unlock()
 
 	if i.ready {
-		return true
+		return nil
 	}
 
 	// Ingester always take at least minReadyDuration to become ready to work
 	// around race conditions with ingesters exiting and updating the ring
 	if time.Now().Sub(i.startTime) < i.cfg.MinReadyDuration {
-		return false
+		return fmt.Errorf("Not ready: waiting for %v after startup", i.cfg.MinReadyDuration)
```

Comment:
Generally considered that errors should be lowercase (https://github.com/golang/go/wiki/Errors).  The argument is that they'll be prefixed by the caller, so you might want to put the `"Not ready: %v"` bit in the HTTP handler?

Reply:
done